sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestArgs.java

```diff
@@ -38,20 +38,20 @@ public final class RntbdRequestArgs {
     private final long nanoTimeCreated;
     private final Stopwatch lifetime;
     private final String origin;
-    private final URI physicalAddress;
+    private final Uri physicalAddressUri;
     private final String replicaPath;
     private final RxDocumentServiceRequest serviceRequest;
     private final long transportRequestId;
 
-    public RntbdRequestArgs(final RxDocumentServiceRequest serviceRequest, final URI physicalAddress) {
+    public RntbdRequestArgs(final RxDocumentServiceRequest serviceRequest, final Uri physicalAddressUri) {
         this.sample = Timer.start();
         this.activityId = serviceRequest.getActivityId();
         this.timeCreated = Instant.now();
         this.nanoTimeCreated = System.nanoTime();
         this.lifetime = Stopwatch.createStarted();
-        this.origin = physicalAddress.getScheme() + "://" + physicalAddress.getAuthority();
-        this.physicalAddress = physicalAddress;
-        this.replicaPath = StringUtils.stripEnd(physicalAddress.getPath(), "/");
+        this.origin = physicalAddressUri.getURI().getScheme() + "://" + physicalAddressUri.getURI().getAuthority();
```

Comment:
NIT three subsequent calls to getURI - instead do this once and cache in local variable

Reply:
maybe even in field of RntbdReqArgs - the conversion is happening over and over again

Reply:
I think this Uri is our class, and getURI() already returns an instance variable, but worth checking.

Reply:
the getURI just returns the instance variable, there is no conversion happening